index.js

```diff
@@ -56,42 +59,19 @@ Client.encoding = Object.freeze({
 function Client (opts) {
   if (!(this instanceof Client)) return new Client(opts)
 
-  this.config(opts)
-
-  Writable.call(this, this._conf)
-
-  const errorproxy = (err) => {
-    if (this.destroyed === false) this.emit('request-error', err)
-  }
-
-  const fail = () => {
-    if (this._writableState.ending === false) this.destroy()
-  }
```

Comment:
Reviewer note: I just moved these down in the same function beside where they are used.